### PR TITLE
[fix] Recursion causing stack overflow when disposing client/service

### LINKED ISSUE #587: [Bug]: Calling Dispose on the Client causes a StackOverflowException
### Software Version

6.7.2

### Language Version

.NET 8

### Operating System

Mac Ventura 13.4

### What happened?

1. Create new instance of the Client class exactly as the README example shows.
2. Call .Dispose().
3. Note that a  StackOverflowException gets thrown.

### What was expected?

I'd expect calling `Dispose` to not loop into a StackOverflowException.

### Sample Code

```csharp
using EasyPost;

namespace Example;

class ExampleClass
{
    static async Task Main()
    {
        var client = new Client(new ClientConfiguration(Environment.GetEnvironmentVariable("EASYPOST_API_KEY")));
        client.Dispose();
    }
}
```

### Relevant logs

```sh
Stack overflow.
Repeat 8025 times:                                   
--------------------------------                     
   at EasyPost.Client.Dispose(Boolean)               
   at EasyPost._base.EasyPostClient.Dispose()        
   at EasyPost._base.EasyPostService.Dispose(Boolean)
   at EasyPost._base.EasyPostService.Dispose()       
--------------------------------                     
   at EasyPost.Client.Dispose(Boolean)               
   at EasyPost._base.EasyPostClient.Dispose()        
   at Example.ExampleClass+<Main>d__0.MoveNext()     
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec
85d7bea7798e]](System.__Canon ByRef)
   at Example.ExampleClass.Main()
   at Example.ExampleClass.<Main>()

Process finished with exit code -1,073,741,571.
```

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

Unhandled recursion during disposal of a `Client` or service would cause a stack overflow.

Closes #587

# Testing

- Add unit tests to verify disposal does not cause stack overflow

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Review by jchen293 — APPROVED** [review]
seems fine, test proves it fixes the issue

File: EasyPost/_base/EasyPostService.cs [hunk 1]
```diff
@@ -66,17 +66,15 @@ public void Dispose()
...
-            }
-
-            // Free native resources (unmanaged objects) and override a finalizer below.
+            // Set the disposed flag to true before disposing of the object to avoid infinite loops
             _isDisposed = true;
+
+            // Dispose managed state (managed objects)
+
+            // Don't dispose of the associated client here, as it may be shared among multiple services
+            // Client.Dispose();
```
Comment (Justintime50) [inline EasyPost/_base/EasyPostService.cs:77]: suggestion: let's not leave commented out code committed.

**Review by Justintime50 — APPROVED** [review]
